kernel/src/simulationTools/Simulation.hpp

```diff
@@ -405,7 +405,7 @@ class Simulation : public std11::enable_shared_from_this<Simulation>
    *  \param m The Model for initializing the OSI.
    *  \param time The current time for initializing the OSI. */
   void prepareIntegratorForDS(SP::OneStepIntegrator osi, SP::DynamicalSystem ds,
-                              SP::Model m=SP::Model(), double time=0);
+                              SP::Model m, double time);
 
```

Comment:
nitpick: we could have Model&m instead of SP::Model.
The use of a SP is unnecessary here. With a reference we also don't have to check whether it is non-empy